Wright_Will/Assignments/Crawler_Part2.py

```diff
@@ -0,0 +1,21 @@
+#crawler Part2
+
+from urllib2 import urlopen
+from bs4 import BeautifulSoup
+import pprint
+
+
+url = 'http://www.codingdojo.com'
+hrefs = {}
+soup = BeautifulSoup(urlopen(url),"html.parser")  #parse website
+for link in soup.find_all('a'):
+    link_str = link.get('href')
+    if link_str in hrefs:
+        hrefs[link_str]+=1
+    else:
+        if link_str[0] =="/":
+            link_str = url + link_str
+        hrefs[link_str] = 1
+
+for each in hrefs:
+    print each + ": " +str(hrefs[each])
```

Comment:
Well done.

If this were a real crawler:

1. You obviously would **not** be running it on coding dojo's network =P
2. You could add a 'depth' parameter so you could call your parsing function recursively. With a max depth threshold.
3. Maybe other filtering options too, such as which domains to dive into. Or alternatively, a blacklist. And perhaps store some other details, such as page title.